abstio/src/abst_data.rs

```diff
@@ -48,9 +48,29 @@ impl Manifest {
                 // Always grab all of these
                 continue;
             }
-            if path.starts_with("data/input/shared") && !data_packs.input.is_empty() {
-                // Grab all of these if the user has opted into any input data at all
-                continue;
+            // If the user has opted into any input data at all, we want to grab some of the shared
+            // input files.
+            if path.starts_with("data/input/shared") {
+                // But some of the files are large, so of course we hardcode some more overrides
+                // here. Maybe some of this data should be scoped to a country, not a city.
+                if path.ends_with("Road Safety Data - Accidents 2019.csv")
+                    || path.ends_with("wu03ew_v2.csv")
+                    || path.ends_with("zones_core.geojson")
+                {
+                    if data_packs.input.iter().any(|x| x.starts_with("gb/")) {
+                        continue;
+                    }
+                } else if path.ends_with("kc_2016_lidar.tif")
+                    || path.ends_with("seattle_contours.geojson")
+                {
+                    if data_packs.input.contains("us/seattle") {
+                        continue;
+                    }
+                } else if !data_packs.input.is_empty() {
+                    // All of the SRTM files land here. Hard to associate them with a country or
```

Comment:
Running the container is currently dominated time-wise by downloading all the SRTM files, because they're not easy to match to a particular place. All of the ad-hoc matching of files with "data packs" is really starting to crumble. :(  But I don't have any better approaches, so continuing to add complexity here.

Reply:
What are the SRTM files? Where are they hosted?

Reply:
They're lower quality, but worldwide-scoped raster elevation data, used by https://github.com/eldang/elevation_lookups, which we call during `importer --raw`. `elevation_lookups` handles downloading missing SRTM files (which ultimately come from https://lpdaac.usgs.gov/products/srtmgl1nv003/, if you're curious). We mirror the ones used by cities we import in `data/input/shared/elevation/srtm`, and if you run the updater with any input data pack filled out, you'll get them all -- currently 139MB uncompressed.

The SRTM filenames have the lat/lon in them; it would be fast, but lots more code, to figure out which ones are used for a particular city. `importer/config/...../foo.poly` has coordinates we could scan. I guess it's a reasonable thing to do later, when the cost of having every VM download 140MB becomes annoying.

Reply:
Thanks for the explanation. I

I would think if the data can be hosted in the same AWS region as your running jobs that the transfer should be less than a second or two (and I think you could avoid paying for any data egress?).

Reply:
Yeah, actually that makes sense. I've been testing by running Docker locally, which is... not in us-east-2. :)  I also kind of suspect the updater buffering all bytes into memory and THEN decompressing will eventually be a bottleneck here that's not too bad to fix.

I'll keep going with the aws batch experiments and see what the download times look like.

Reply:
> The SRTM filenames have the lat/lon in them; it would be fast, but lots more code, to figure out which ones are used for a particular city.

Sounds like the government could used some cloud-optimized geotifs 🙃